runtime/sema/type.go

```diff
@@ -7070,3 +7074,245 @@ func isNumericSuperType(typ Type) bool {
 
 	return false
 }
+
+// EntitlementType
+
+type EntitlementType struct {
```

Comment:
On one hand it's good that entitlements are defined as types, i.e. are in the type namespace, which prevents an entitlement and a type to be named the same and avoids confusion. On the other hand it also seems a bit odd that an entitlement can then be used as a type, e.g. for a variable, parameter, return type, etc. Maybe that should be explicitly forbidden to avoid confusion for the developer?

Reply:
Yes this is already explicitly forbidden (see the `DirectEntitlementAnnotationError` error).

Reply:
Nice!